### PR TITLE
Added search by Collections to site wide search.

### PR DESCRIPTION
**Pull request by damisul**
Also:
- renamed ManifestationsSearch to SiteWideSearch
- added automatic reindexing of DictionaryEntry records

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -148,6 +148,7 @@ group :development do
 end
 
 group :test, :development do
+  gem 'byebug'
```
Comment (damisul) [inline Gemfile:151]: I use this gem for debugging

File: app/controllers/search_controller.rb [hunk 2]
```diff
@@ -7,11 +7,8 @@ def index
   def results
     begin
       @searchterm = params[:search].nil? ? sanitize_term(params[:q]) : sanitize_term(params[:search])
-      unless params[:search].nil?
-        @search = ManifestationsSearch.new(@searchterm)
-      else
-        @search = ManifestationsSearch.new(query: @searchterm)
-      end
+      @search = params[:search].present? ? SiteWideSearch.new(@searchterm) : SiteWideSearch.new(query: @searchterm)
```
Comment (damisul) [inline app/controllers/search_controller.rb:10]: I did not found any places where 'search' param is passed here. So perhaps we can remove this code branch?

File: app/lib/site_wide_search.rb [hunk 3]
```diff
@@ -1,7 +1,8 @@
 # frozen_string_literal: true
 
 require 'multi_index_search_request'
-class ManifestationsSearch
+# Class representing request to search whole website (texts, people, etc.) by string query
+class SiteWideSearch
```
Comment (damisul) [inline app/lib/site_wide_search.rb:5]: I've renamed class as its name was misleading

File: app/lib/site_wide_search.rb [hunk 4]
```diff
@@ -101,8 +108,4 @@ def tags_filter
   def highlight
     index.highlight(max_analyzed_offset: 999_000, fields: { fulltext: {}, deftext: {} })
   end
-
```
Comment (damisul) [inline app/lib/site_wide_search.rb:104]: This code was add probably to resolve items ordering issue, but after we implemented script-based ordering it become obsolete, so I've removed it

File: app/views/search/results.html.haml [hunk 5]
```diff
@@ -27,6 +27,9 @@
             - unless result._data.nil? or result._data['highlight'].nil?
               %br
               != result._data['highlight']['fulltext'].join("<br />").gsub('<em>','<b>').gsub('</em>','</b>')
+          - elsif result.instance_of?(CollectionsIndex)
+            %b= Collection.model_name
+            %b= link_to([result.title, result.subtitle].compact_blank.join(' / '), collection_path(result.id))
```
Comment (damisul) [inline app/views/search/results.html.haml:32]: Probably we may need to add there some tweaks for right-to-left text orientation. Also not sure if subtitle should be displayed at all. I simply did some basic view, we can update later.

File: Gemfile (hunk 1 above)
Comment (abartov) [reply Gemfile:151]: So did I, but the latest and recommended debugging gem is 'debug'.

File: spec/controllers/search_controller_spec.rb [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+      create(:authority, name: 'Test Authority')
+      create(:manifestation, title: 'Test Manifestation')
+      create(:collection, title: 'Test Collection')
+    end
+  end
+
+  describe '#results' do
+    subject { get :results, params: { q: 'Test' } }
+
+    it { is_expected.to be_successful }
```
Comment (abartov) [inline spec/controllers/search_controller_spec.rb:18]: Please add testing of result count and order.

**Review by abartov — CHANGES_REQUESTED** [review]
see inline comments.

File: app/controllers/search_controller.rb (hunk 2 above)
Comment (abartov) [reply app/controllers/search_controller.rb:10]: Could be a remnant of some older code, yes.

**Review by abartov — APPROVED** [review]

**Pull request merged**